### DISCUSSION TITLE
Few questions to the new app directory

### DISCUSSION THREAD

**Discussion by serhat-m** [vercel/next.js#45470] [Help] [answered]
Hi everyone, I have a few questions about the new app directory in Next.js 13.

- In the "old" pages directory, you could manipulate the data for the HTML head with the next/head component within the corresponding page. If I retrieved data on the page that might be relevant to the title of the page, I could also use this in the "head" component. In the app directory, the HTML head is isolated in the appropriate head.js file. This may mean that I would have to request the same data set twice. Is there a more elegant solution to this that I may be overlooking?

With the "not-found.js" file and the "notFound" function, it is easy to catch errors:
- How can I configure the HTML head of the error message?
- When I visit a page that is not present in my routing structure, the default 404 page is returned (let's say /blog is present but the user requests /bloggg). I cannot execute the "notFound" function on a non-existent route. What are the options here?

Thanks in advance for the replies.

**Comment by ciruz** [comment, marked as answer]
Hey,

1. Yes, you can and have to fetch the same url twice, for ``head.js`` and ``page.js``, this is the recommended way, so it's the solution to go, for now, but don't worry the requests gets cached.
> When rendering a route, Next.js will [automatically dedupe requests](https://beta.nextjs.org/docs/data-fetching/fundamentals#automatic-fetch-request-deduping) for the same data across layout.js, page.js, and head.js.
- https://beta.nextjs.org/docs/api-reference/file-conventions/head#data-fetching-and-dynamic-tags
> If you need to fetch the same data (e.g. current user) in multiple components in a tree, Next.js will automatically cache fetch requests that have the same input in a temporary cache. This optimization is called deduping (or deduplication), and prevents the same data from being fetched more than once during a rendering pass.
- https://beta.nextjs.org/docs/data-fetching/fundamentals#automatic-fetch-request-deduping

2 & 3. the current notFound state is more of a temporary solution. From the docs: https://beta.nextjs.org/docs/upgrade-guide#migrating-from-pages-to-app
> pages/404.js has been temporarily replaced with [notFound()](https://beta.nextjs.org/docs/api-reference/file-conventions/not-found) for specific route segments. We are also working on a new 404.js file to handle global errors.
